seaport-subgraph/src/modules/airstack/nft.ts

```diff
@@ -0,0 +1,229 @@
+
+import {
+    AirAccount,
+    AirNftTransaction,
+    AirToken,
+    AirBlock,
+    AirNftSaleRoyalty
+} from "../../../generated/schema";
+
+export namespace nft {
+    export function trackNFTSaleTransactions(
+        txHash: string,
+        txIndex: BigInt,
+        NftSales: Sale[],
+        protocolType: string,
+        protocolActionType: string,
+        timestamp: BigInt,
+        blockHeight: BigInt,
+        blockHash: string
+    ): void{
+        if (NftSales.length == 0) {
+            return;
+        }
+
+
+        let chainID = 1;
+        let block = getOrCreateAirBlock(chainID.toString()+"-"+blockHeight.toString());
+        block.number = blockHeight;
+        block.timestamp = timestamp;
+        block.hash = blockHash;
+        
+        let transactionCount = NftSales.length;
+        for(let i=0; i<transactionCount; i++){
+            // Payment Token
+            let paymentToken = getOrCreateAirToken(NftSales[i].paymentToken.toHexString());
+            paymentToken.save();
+
+            // Account
+            let buyerAccount = getOrCreateAirAccount(NftSales[i].buyer.toHexString());
+            buyerAccount.createdAt = block.id
+            let sellerAccount = getOrCreateAirAccount(NftSales[i].seller.toHexString());
+            sellerAccount.createdAt = block.id
+            
+            let royalties = new Array<string>();
+            
+            let feeAccount = getOrCreateAirAccount(NftSales[i].protocolFeesBeneficiary.toHexString());
+            feeAccount.createdAt = block.id
+
+            // Sale Token
+            let saleToken = getOrCreateAirToken(
+                NftSales[i].nft.collection.toHexString()
+            );
+            saleToken.save();
+            
+            // Transaction
+            let transactionId = getNFTSaleTransactionId(
+                txHash,
+                NftSales[i].nft.collection.toHexString(),
+                NftSales[i].nft.tokenId
+            )
+            
+            let transaction = AirNftTransaction.load(transactionId);
+            if (transaction != null){
+                if (transaction.from == Address.zero().toHexString()){
+                    if (buyerAccount.address != Address.zero().toHexString()){
+                      transaction.from = buyerAccount.id;
+                    } else if (sellerAccount.address != Address.zero().toHexString()){
+                      transaction.from = sellerAccount.id;
+                    }
+                  }
+                  else if (transaction.to == Address.zero().toHexString()){
+                    if (buyerAccount.address != Address.zero().toHexString()){
+                        transaction.to = buyerAccount.id;
+                    } else if (sellerAccount.address != Address.zero().toHexString()){
+                        transaction.to = sellerAccount.id;
+                    }
+                  }
+            }else{
+                transaction = getOrCreateAirNftTransaction(
```

Comment:
What's the case when you can get the same id?

Reply:
In seaport the orderFulfilled event is being emitted two times. It helps to get additional details.